File: Som.Net/Som/Data/DirectDataProvider.cs
using System;
using System.Collections.Generic;

namespace Som.Data
{
    public class DirectDataProvider : ILearningDataPersister
    {
        public List<List<double>> Data { get; private set; }

        public DirectDataProvider(List<List<double>> data)
        {
            Data = data;
        }

        public List<List<double>> GetData()
        {
            return Data;
        }

        public void SaveData(List<List<double>> data)
        {
            throw new NotImplementedException();
        }
    }
}

File: trunk/JavaSom/Data/DirectDataProvider.java
package Data;

import sun.reflect.generics.reflectiveObjects.NotImplementedException;

public class DirectDataProvider implements ILearningDataPersister
	{
		private java.util.ArrayList<java.util.ArrayList<Double>> privateData;
		public final java.util.ArrayList<java.util.ArrayList<Double>> getData()
		{
			return privateData;
		}
		private void setData(java.util.ArrayList<java.util.ArrayList<Double>> value)
		{
			privateData = value;
		}

		public DirectDataProvider(java.util.ArrayList<java.util.ArrayList<Double>> data)
		{
			setData(data);
		}

		public final java.util.ArrayList<java.util.ArrayList<Double>> GetData()
		{
			return getData();
		}

		public final void SaveData(java.util.ArrayList<java.util.ArrayList<Double>> data)
		{
			throw new NotImplementedException();
		}
	}